lib/core/src/user_administration.cpp

```diff
@@ -0,0 +1,486 @@
+#include "user_administration.hpp"
+
+#undef rxGeneralAdmin
+
+#ifdef IRODS_USER_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+    #include "rsGeneralAdmin.hpp"
+    #define rxGeneralAdmin rsGeneralAdmin
+    #define RODS_SERVER 1 // For irods::query<rsComm_t>
+#else
+    #include "generalAdmin.h"
+    #define rxGeneralAdmin rcGeneralAdmin
+#endif // IRODS_USER_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include "obf.h"
+#include "authenticate.h"
+#include "irods_query.hpp"
+#include "query_builder.hpp"
+
+#include <array>
+#include <iostream>
+
+namespace irods::experimental::administration::NAMESPACE_IMPL
+{
+    inline namespace v1
+    {
+        namespace
+        {
+            auto to_user_type(std::string_view user_type_string) -> user_type
+            {
+                // clang-format off
+                if      (user_type_string == "rodsuser")  { return user_type::rodsuser; }
+                else if (user_type_string == "rodsadmin") { return user_type::rodsadmin; }
```

Comment:
groupadmin?

Reply:
I haven't added support for groupadmin. Is that something that needs to be managed by this library?

Reply:
doesn't have to be - wasn't sure if it was an oversight.   perhaps leave a note that maybe in the future?

Reply:
Sure. If it's straight forward to implement, I may go on and add it. But I'll leave a comment for sure.